### ISSUE TITLE
[FEAT] Test new executable to get rid of java and improve performance

### ISSUE THREAD

**Issue by theotherp** [linuxserver/docker-nzbhydra2#29]
### Is this a new feature request?

- [X] I have searched the existing issues

### Wanted change

I'm currently working on a major update for NZBHydra. If all goes well I will be able to release compiled binaries instead of java libraries, meaning that you don't need to have java installed. Everything you need is in the binary. Apart from the hassle of needing java this also promises some drastic improvements on startup time and memory usage.

One problem I encountered is that the binary is not a "static image" but a "mostly static image", e.g. it needs glibc to run which is why an alpine base image won't work. I've tried the steps described here: https://wiki.alpinelinux.org/wiki/Running_glibc_programs but didn't have any luck. Instead I was able to get it to run using an ubuntu base image and installing libfreetype6.

Apart from that the docker file should mostly look the same. The python wrapper will still be included and just run a binary instead of java.

I've made a minimal example here: https://ufile.io/mzbsci6q Please do not release this as it's just a test version.

The proposed next steps:

- You test it and see if it works. If you're able to get it to run on alpine or wherever, great.
- I still need some time (weeks) to test everything properly and make sure that the new code works as expected.
- I'll then let you know when everything is ready. I don't know how your pipeline works but hydra version 4.x will be the old way (java) and 5.x will be new one so you could check for that to make the transition work.

Thanks!

### Reason for change

See above.

**Comment by thespad** [comment]
Both images (latest and Dev) are on Ubuntu at the moment because Alpine don't do JRE 9+ packages for 32bit arm any more, so that shouldn't be a problem.

We'll have a play with the new build and see what the best option is - for releasing as we'll need to change a decent chunk of the Dockerfile and service logic, we'll get the PRs ready to go and just merge them after you've made the public release.

Are you planning to do Dev first and then stable a while after or work on v5 outside of that and bump both branches at the same time?

**Comment by theotherp** [comment]
I'll probably do dev first. I don't expect everything to go completely smoothly, to be honest. Probably dev first and hopefulle main a couple of days later.

**Comment by thespad** [comment]
Are there meaningful changes to the config .yml for v5? When I copied the existing default it complained it was corrupt.

Other than that it seems to build and run OK copying the zip in locally.

**Comment by theotherp** [comment]
No. Validate it using an online validator, it should be easy to fix. Please
let me know what was wrong so I can fix it. I tried migration on several
instances on windows and Linux and it always worked fine.

**Comment by thespad** [comment]
Oh, I know why, I'm an idiot. The changes I made to the Dockerfile for testing mean it was writing a 404 response to the config file that was getting copied.

**Comment by thespad** [comment]
OK, build and run looks good. Just need to know once you've published a (pre-)release, so we can validate downloading it from GH rather than copying the zip by hand.

If you want to have a look I've pushed it to `docker.io/thespad/playground:nzbhydra2`, just note that it's ~80Mb bigger than the real one would be because the zip gets copied in its own layer so can't be deleted from the image without multi-staging the build, which is overkill for a test.

**Comment by theotherp** [comment]
Sorry, I didn't get a notification for your comment for some reason.

I've finished all my tests and all my tests are executed successfully. I'm ready to release but plan to wait until the weekend after the next (around 20th).

Since my post I realised that the container still needs java in order to migrate from the last version. When everybody has migrated (whenever that may be) we can get rid of java once and for all.

This is a new zip with a new executable: https://ufile.io/drby73fu

I was unabled to start your container, the config file actually contains a 404 for me as well.

Thanks.
